woocommerce/woocommerce-android
# display SKU on scan failure snack bar

Closes: #9295 


### Description


This PR changes the required returning type of the message on `OnAddingProductViaScanningFailed` class (and related methods) to String so we can have a dynamic error message in the snack bar. 

### Testing instructions


1. Go to the orders screen
2. click on the option to scan a barcode
3. scan an invalid barcode like the one here https://barcode.tec-it.com/en/?data=123_invalid_sku
4. Note the message saying the SKU `123_invalid_sku` is invalid
5. Feel free to change the code in the link to make sure it's updated in the snack bar when you scan again. 

### Images/gif


| Before | After |
| ---- | ---- |
| <img src="https://github.com/woocommerce/woocommerce-android/assets/30724184/7447cf2b-0773-41c1-af5e-5a82c3d364bf" width="350"/> | <img src="https://github.com/woocommerce/woocommerce-android/assets/30724184/d4e84271-c5bb-47f5-9335-ada22fb68db0" width="350"/> |
